src/main/java/com/facebook/openwifirrm/modules/Modeler.java

```diff
@@ -94,6 +96,10 @@ public static class DataModel {
 			new ConcurrentHashMap<>();
 
 		/** List of latest state per device. */
+		public Map<String, List<State>> latestStates =
+			new ConcurrentHashMap<>();
+
+		/** The most latest state per device.*/
 		public Map<String, State> latestState = new ConcurrentHashMap<>();
```

Comment:
wondering if it's cleaner to just have one state mapping, and just reference the last element in the list vs having two separate state mapping. This would also necessitate that the list become thread safe as well though so I'm not sure :)

Reply:
That's also one of my questions. Know that you also use the `latestState` filed in #61, would you prefer only keeping `latestStates`? The `latestStates` will keep all the States in the Kafka record without any time frame. Do you have any suggestions on thread-safe?

Reply:
I think it makes sense to only keep one, since one of them would essentially be redundant data. 

For thread safety, I'm not so sure since I'm still pretty new to Java - not sure if there's a concurrent list

Reply:
Sorry that I think I should have specified it clearly. Both need to be kept. We use the latest one for the optimization algorithm and we use the aggregated one for the root cause analysis. The purpose of this one is a little bit different than the wifiscan aggregation.

Reply:
@pohanhf I think this thread is more about whether we need two variables. We will still have the latest state as the last element in `latestStates` but Jeffrey and I are saying that we don't need a separate `latestState` variable (we can instead to `latestStates.get(0)`

Reply:
Ah got it. Thanks for the clarification. :)